### ISSUE TITLE
Duplicate lookup errors

### ISSUE THREAD

**Issue by ilya-biryukov** [llvm/llvm-project#139226]
See https://gcc.godbolt.org/z/jo8Mv6838
```cpp
struct Foo {
    template<class T>
    void LookupWithID();
};

void test(Foo *f) {
    f->LookupWithId<Info>(); // Note: ID is misspelled as Id
}
```

Expected: one lookup error about LookupWithId.
Actual: two duplicate errors `no template named 'LookupWithId' in 'Foo'; did you mean 'LookupWithID'?`

Could this be happening because we parse tentatively twice and report the errors both times?

**Comment by shafik** [comment]
This goes back to clang-3.1: https://godbolt.org/z/jMsa37WKj

**Comment by shafik** [comment]
We had this report but this case is fixed: https://github.com/llvm/llvm-project/issues/24786

**Comment by ilya-biryukov** [comment]
Thanks a lot for a quick fix!
